docs/core/testing/unit-testing-fsharp-with-dotnet-test.md

```diff
@@ -0,0 +1,169 @@
+to add the class library project to the solution. Next, create the *MathService.Tests* directory. The following outline shows the directory structure:
+
+```
+/unit-testing-with-fsharp
+    unit-testing-with-fsharp.sln
+    /MathService
+        Source Files
+        MathService.fsproj
+    /MathService.Tests
+```
+
+Make the *MathService.Tests* directory the current directory and create a new project using [`dotnet new xunit -lang F#`](../tools/dotnet-new.md). This creates a test project that uses xUnit as the test library. The generated template configures the test runner in the *MathServiceTests.fsproj*:
+
+```xml
+<ItemGroup>
+  <PackageReference Include="Microsoft.NET.Test.Sdk" Version="15.3.0-preview-20170628-02" />
+  <PackageReference Include="xunit" Version="2.2.0" />
+  <PackageReference Include="xunit.runner.visualstudio" Version="2.2.0" />
+</ItemGroup>
+```
+
+The test project requires other packages to create and run unit tests. `dotnet new` in the previous step added xUnit and the xUnit runner. Now, add the `MathService` class library as another dependency to the project. Use the [`dotnet add reference`](../tools/dotnet-add-reference.md) command:
+
+```
+dotnet add reference ../MathService/MathService.fsproj
+```
+
+Another option is to edit the *PrimeService.Tests.fsproj* file. Directly under the first `<ItemGroup>` node, add another `<ItemGroup>` node with a reference to the library project:
+
+```xml
+<ItemGroup>
+  <ProjectReference Include="..\MathService\MathService.fsproj" />
+</ItemGroup>
+```
+
+You can see the entire file in the [samples repository](https://github.com/dotnet/docs/blob/master/samples/core/getting-started/unit-testing-with-fsharp/MathService.Tests/MathService.Tests.fsproj) on GitHub.
+
+You have the following final solution layout:
+
+```
+/unit-testing-with-fsharp
+    unit-testing-with-fsharp.sln
+    /MathService
+        Source Files
+        MathService.fsproj
+    /MathService.Tests
+        Test Source Files
+        MathServiceTests.fsproj
+```
+
+## Creating the first test
+
+Before building the library or the tests, execute [`dotnet sln add .\PrimeService.Tests\PrimeService.Tests.csproj`](../tools/dotnet-restore.md) in the *unit-testing-using-dotnet-test* directory. 
+
+The TDD approach calls for writing one failing test, making it pass, then repeating the process. Open *Tests.fs* and add the following code:
+
+```fsharp
+[<Fact>]
+let ``My test`` () =
+    Assert.True(true)
+
+[<Fact>]
+let ``Fail every time`` () = Assert.True(false)
+```
+
+The `[<Fact>]` attribute denotes a method as a single test. From the *unit-testing-with-fsharp*, execute [`dotnet test`](../tools/dotnet-test.md) to build the tests and the class library and then run the tests. The xUnit test runner contains the program entry point to run your tests. `dotnet test` starts the test runner using the unit test project you've created.
```

Comment:
"denotes a method as a single test" is unclear -- I find "single test" misleading. The attributes denotes a particular method as a test method that's to be run by the test framework. Non-marked methods aren't run unless they're called from a test method. I'd change this to "denotes a test method that is be run by the test framework." This applies as well to the `[TestMethod]` attribute